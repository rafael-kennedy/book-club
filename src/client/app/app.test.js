import { shallowMount } from "@vue/test-utils";
import subject from "./app.vue";

describe("main app component", () => {
  test("displays welcome message when not logged in", () => {
    const wrapper = shallowMount(subject);
    expect(wrapper).toMatchSnapshot();
  });

  test("displays tabs component when logged in", () => {
    const wrapper = shallowMount(subject);
    debugger;
    wrapper.vm.user = { email: "fake" };
    expect(wrapper).toMatchSnapshot();
  });
});
