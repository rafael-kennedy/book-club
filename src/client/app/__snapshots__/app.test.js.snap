// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main app component displays tabs component when logged in 1`] = `
<div>
  <ui-snackbar-container-stub queue-snackbars="true" position="right"></ui-snackbar-container-stub>
  <ui-toolbar-stub remove-brand-divider="" text-color="white" type="colored" background-color="primary">
    Book Club App
    <template></template></ui-toolbar-stub>
  <ui-tabs-stub type="text">
    <ui-tab-stub title="Nominate Books"></ui-tab-stub>
    <ui-tab-stub title="Vote"></ui-tab-stub>
    <ui-tab-stub title="Upcoming Meetings"></ui-tab-stub>
  </ui-tabs-stub>
</div>
`;

exports[`main app component displays welcome message when not logged in 1`] = `
<div>
  <ui-snackbar-container-stub queue-snackbars="true" position="right"></ui-snackbar-container-stub>
  <ui-toolbar-stub remove-brand-divider="" text-color="white" type="colored" background-color="primary">
    Book Club App
    <template></template></ui-toolbar-stub>
  <div>
    <h1>Welcome to a generic book club app!</h1>
    <p>Please log in by clicking the icon above to get started!</p>
  </div>
</div>
`;
